data/service/snapshot-service.js

```diff
@@ -111,30 +111,25 @@ exports.SnapshotService = Montage.specialize(/** @lends SnapshotService# */ {
         value: function(a, b) {
             var result = a === b;
             if (!result) {
-                if (typeof a === "object" && typeof b === "object") {
-                    
-                    // TODO WTF
-                    result = (!!a === !!b);
```

Comment:
;)

This checks if a and b have the same truthy values, but doing it this way isn't necessary. We already know from previous conditions that `a !== b` and `a !== undefined` and `b !== undefined`. The next part of the function uses Object.keys(), which needs a truthy (non-null) parameter, so we just need to check that a and b are both truthy. If they are both falsy objects then the `a === b` condition on the first line would have been true already.